File: backend/src/services/task_service.py
from repo.task_repo import TaskRepo

class TaskService():

    def create_task(self, task_data):
        return TaskRepo().create_task(task_data)
    
    def get_taskById(self, task_id):
        return TaskRepo().get_taskById(task_id)
    
    def delete_task(self, task_id):
        return TaskRepo().delete_task(task_id)
    
    def update_task(self, task_id, task_data):
        return TaskRepo().update_task(task_id, task_data)
    
    def get_user_tasks(self):
        return TaskRepo().get_user_tasks()

File: frontend/src/api/taskService/taskService.ts
import { axiosInstance } from "@/pages/config/axios";
import type { TaskStats } from "@/pages/dashboard/types";
import type { Tasks } from "./types";

enum TaskService {
  base = `/api/task`,
}

const getMetricsUrl = () => `${TaskService.base}/dashboard`;

const getTasksUrl = () => `${TaskService.base}/tasks`;

const createTaskUrl = () => `${TaskService.base}/create`;

const updateTaskUrl = (taskId: string) => `${TaskService.base}/${taskId}`;

export const getMetrics = async (): Promise<TaskStats> => {
  const response = await axiosInstance.get<TaskStats>(getMetricsUrl());
  return response.data;
};

export const getTasks = async (payload: Object): Promise<Tasks> => {
  const response = await axiosInstance.post<Tasks>(getTasksUrl(), payload);
  return response.data;
};

export const createTask = async (payload: Object) => {
  const response = await axiosInstance.post(createTaskUrl(), payload);
  return response.data;
};

export const updateTask = async (taskId: string, payload: Object) => {
  const response = await axiosInstance.put(updateTaskUrl(taskId), payload);
  return response.data;
}
